src/Container.php

```diff
@@ -283,7 +284,15 @@ protected function getServiceInstance($service)
     {
         // does the definition exist?
         if (! $this->has($service)) {
-            throw Exception::serviceNotFound($service);
+            if ($this->resolver instanceof AutoResolver) {
+                try {
+                    $this->services[$service] = $this->newInstance($service);
+                } catch (Exception $e) {
+                    throw Exception::serviceNotFound($service);
```

Comment:
@harikt It's closer, yes.

1. `has` should return `true` if the class exists and is instantiable (if service not defined && auto-resolve is true).
2. `getServiceInstance` should use `newInstance` after checking the delegateContainer, and should not catch any exceptions.

However, I'm waiting for input from [container-interop](https://github.com/container-interop/container-interop/pull/65#issuecomment-185217599) on this matter.

Thanks for your help so far. =)